src/Core/Search/Discussion/Gambit/AuthorGambit.php

```diff
@@ -50,7 +51,7 @@ protected function conditions(AbstractSearch $search, array $matches, $negate)
 
         $ids = [];
         foreach ($usernames as $username) {
-            $ids[] = $this->users->getIdForUsername($username);
+            $ids[] = is_numeric($username) ? $username : $this->users->getIdForUsername($username);
```

Comment:
This PR is applicable only if we decide that username can not be digits only. As I mentioned above, like the current user API, it won't work properly in certain cases.